tidb-server/main.go

```diff
@@ -75,6 +75,7 @@ var (
 	queryLogMaxlen      = flag.Int("query-log-max-len", 2048, "Maximum query length recorded in log")
 	startXServer        = flagBoolean("xserver", false, "start tidb x protocol server")
 	tcpKeepAlive        = flagBoolean("tcp-keep-alive", false, "set keep alive option for tcp connection.")
+	planCacheSize       = flag.Int("cache-size", 100, "the maximum number of plans to be cached for each session.")
```

Comment:
Thank you for the sugguestions. 

(1)  I've changed the codes as follows:
`-  planCache *cache.UnsafeLRUCache`
`+  preparedPlanCache *cache.UnsafeLRUCache`
` `
`   sessionVars    *variable.SessionVars`
`   sessionManager util.SessionManager`
`@@ -185,7 +185,7 @@ func (s *session) SetConnectionID(connectionID uint64) {`
` }`
` `
` func (s *session) Plan() *cache.UnsafeLRUCache {`
`-  return s.planCache`
`+  return s.preparedPlanCache`
` }`
` `
` func (s *session) SetSessionManager(sm util.SessionManager) {`
`@@ -840,8 +840,8 @@ func (s *session) Close() {`
`   if s.statsCollector != nil {`
`       s.statsCollector.Delete()`
`   }`
`-  if s.planCache != nil {`
`-      defer s.planCache.Clear()`
`+  if s.preparedPlanCache != nil {`
`+      defer s.preparedPlanCache.Clear()`
`   }`
`   if err := s.RollbackTxn(); err != nil {`
`       log.Error("session Close error:", errors.ErrorStack(err))`
`@@ -1007,10 +1007,10 @@ func createSession(store kv.Storage) (*session, error) {`
`       return nil, errors.Trace(err)`
`   }`
`   s := &session{`
`-      planCache:   cache.NewUnsafeLRUCache(int64(planCacheSize)),`
`+      preparedPlanCache: cache.NewUnsafeLRUCache(int64(preparedPlanCacheSize)),`
`   }`

(2) I'v added TODO as follows:
`-  planCacheSize       = flag.Int("cache-size", 100, "the maximum number of plans to be cached for each session.")`
`+  //TODO: preparedPlanCacheSize parameter will be added to session or global variables so that can adjust the plan cache size without restarting their service at any time`
`+  preparedPlanCacheSize = flag.Int("prepared-plan-cache-size", 100, "the maximum number of plans to be cached for each session.")`